trackpy/feature.py

```diff
@@ -762,11 +787,11 @@ def batch(frames, diameter, minmass=100, maxsize=None, separation=None,
                      filter_before=filter_before, filter_after=filter_after)
 
     if meta:
-        if isinstance(meta, str):
-            filename = meta
+        if isinstance(meta, six.string_types):
+            with open(filepath_or_obj, 'w') as file_obj:
+                record_meta(meta_info, file_obj)
         else:
-            filename = 'feature_log_%s.yml' % timestamp
-        record_meta(meta_info, filename)
+            record_meta(meta_info, file_obj)
```

Comment:
Should'n this be `file_obj -> `meta` as `file_obj` only exists if the above context manager gets hit.